reference/docs-conceptual/dev-cross-plat/resolving-dependency-conflicts.md

```diff
@@ -857,157 +769,27 @@ demonstrates how to migrate a module to use an ALC, while keeping that module wo
-
-            // First we want to know if this is the top-level assembly
-            if (asmName.Name.Equals("LoadFileModule.Engine"))
-            {
-                s_engineLoaded = true;
-                return Assembly.LoadFile(Path.Combine(s_dependenciesDirPath, "Unrelated.Engine.dll"));
-            }
+This solution is also used by the [Bicep PowerShell module](https://github.com/PSBicep/PSBicep),
+and the blog post [Resolving PowerShell Module Conflicts](https://pipe.how/get-assemblyloadcontext/)
+is another good read about this solution.
 
-            // Otherwise, if that assembly has been loaded, we must try to resolve its dependencies too
-            if (s_engineLoaded
-                && s_dependencies.TryGetValue(asmName.Name, out int requiredMajorVersion)
-                && asmName.Version.Major == requiredMajorVersion)
-            {
-                string asmPath = Path.Combine(s_dependenciesDirPath, $"{asmName.Name}.dll");
-                return Assembly.LoadFile(asmPath);
-            }
+### Assembly resolving handler for side-by-side loading
 
-            return null;
-        }
-    }
-}
-```
+Although being robust, the solution described above requires the module assembly to not directly reference the dependency assemblies,
+but instead, to reference a wrapper assembly which then references the dependency assemblies.
+The wrapper assembly acts like a bridge, forwarding the calls from the module assembly to the dependency assemblies.
+This makes it usually a non-trivial amount of work to adopt this solution:
 
-Finally, the layout of the module is also similar to the ALC module:
+- For a new module, this would add additional complexity to the design and implementaion
+- For an existing module, this would require significant refactoring
 
-```
-LoadFileModule/
-  + LoadFileModule.Cmdlets.dll
-  + LoadFileModule.psd1
-  + Dependencies/
-  |  + LoadFileModule.Engine.dll
-  |  + CsvHelper.dll
-```
+There is a simplified solution to achieve side-by-side assembly loading, by
+- Hooking up a `Resolving` event to a custom Assembly Load Context
+- Or, hooking up a `Resolving` event to `Assembly.LoadFile()`
 
-With this structure in place, **LoadFileModule** now supports being loaded alongside other modules
-with a dependency on **CsvHelper**.
-
-Because our handler applies to **all** `AssemblyResolve` events across the Application Domain,
-we need to make some specific design choices here:
-
-- To narrow the window in which our event may interfere with other loading, we only start handling
-  general dependency loading after `LoadFileModule.Engine.dll` has been loaded.
-- We push `LoadFileModule.Engine.dll` out into the Dependencies directory, so that it's loaded by a
-  `LoadFile()` call rather than by PowerShell. This means its own dependency loads always raise an
-  `AssemblyResolve` event, even if another `CsvHelper.dll` (for example) is loaded in PowerShell.
-  This gives us the opportunity to find the correct dependency.
```

Comment:
This is actually not true. Even if having `LoadFileModule.Engine.dll` loaded by `Assembly.LoadFile`, the loading requests for its dependencies will still first go through the default `AssemblyLoadContext` before hitting `AssemblyResolve` event, and thus if a higher version of the same assembly is already loaded in the default ALC, that assembly will be used to serve the request. See details in [the loading order doc](https://github.com/dotnet/docs/blob/main/docs/core/dependency-loading/loading-managed.md#algorithm).

So, it's really meaningless to have a wrapper assembly for this solution.